kratos/solving_strategies/strategies/arc_length_strategy.h

```diff
@@ -0,0 +1,693 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ \.
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Alejandro Cornejo
+//                   Ignasi Pouplana
+//
+
+#if !defined(KRATOS_ARC_LENGTH_STRATEGY)
+#define KRATOS_ARC_LENGTH_STRATEGY
+
+// System includes
+#include <iostream>
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "solving_strategies/strategies/residualbased_newton_raphson_strategy.h"
+#include "solving_strategies/convergencecriterias/convergence_criteria.h"
+#include "utilities/builtin_timer.h"
+
+//default builder and solver
+#include "solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h"
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/**
+ * @class ArcLengthStrategy
+ * @ingroup KratosCore
+ * @brief This is the base ArcLengthStrategy
+ * @details The theoretical details can be found in "Geometrical interpretation of the ARC-LENGTH method", M. Fafard and B. Massicotte,
+ * Computers and Structures Vol 46 pp 603-615 (1993). (Ramm arc-length method)
+ * @author Alejandro Cornejo
+ */
+template <class TSparseSpace, class TDenseSpace, class TLinearSolver>
+class ArcLengthStrategy
+    : public ResidualBasedNewtonRaphsonStrategy<TSparseSpace, TDenseSpace, TLinearSolver>
+{
+    public:
+    ///@name Type Definitions
+    ///@{
+
+    // Counted pointer of ClassName
+    KRATOS_CLASS_POINTER_DEFINITION(ArcLengthStrategy);
+
+    typedef ResidualBasedNewtonRaphsonStrategy<TSparseSpace, TDenseSpace, TLinearSolver> BaseType;
+    typedef typename BaseType::TBuilderAndSolverType TBuilderAndSolverType;
+    typedef typename BaseType::TDataType TDataType;
+    typedef TSparseSpace SparseSpaceType;
+    typedef typename BaseType::TSchemeType TSchemeType;
+    typedef typename BaseType::DofsArrayType DofsArrayType;
+    typedef typename BaseType::TSystemMatrixType TSystemMatrixType;
+    typedef typename BaseType::TSystemVectorType TSystemVectorType;
+    typedef typename BaseType::LocalSystemVectorType LocalSystemVectorType;
+    typedef typename BaseType::LocalSystemMatrixType LocalSystemMatrixType;
+    typedef typename BaseType::TSystemMatrixPointerType TSystemMatrixPointerType;
+    typedef typename BaseType::TSystemVectorPointerType TSystemVectorPointerType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    // Constructor
+    ArcLengthStrategy(
```

Comment:
Can you add a "pure" parameters constructor?, see NR as example

Reply:
for what it is used? curiosity only hehe

Reply:
for the factories

In fact can you please remove this regular CTor with many individual settings and only add the parameters ctor?